components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/ImportUtils.java

```diff
@@ -791,6 +816,96 @@ private static JsonElement retrieveValidatedDTOObject(String pathToArchive, Bool
+
+            String pathToJ2File = pathToArchive + fileName + ImportExportConstants.SYNAPSE_EXTENSION;
+
+            String synapsePolicyDefinitionJsonContent = readDefinitionFiles(pathToJ2File);
+
+            if (synapsePolicyDefinitionJsonContent != null) {
+                synapseDefinition = new OperationPolicyDefinition();
+                synapseDefinition.setContent(synapsePolicyDefinitionJsonContent);
+                synapseDefinition.setGatewayType(OperationPolicyDefinition.GatewayType.Synapse);
+                synapseDefinition.setMd5Hash(APIUtil.getMd5OfOperationPolicyDefinition(synapseDefinition));
+                operationPolicyData.setSynapsePolicyDefinition(synapseDefinition);
+            }
+
+            String pathToChoreoFile = pathToArchive + fileName + ImportExportConstants.CHOREO_CONNECT_EXTENSION;
+
+            String choreoConnectPolicyDefinitionJsonContent = readDefinitionFiles(pathToChoreoFile);
+
+            if (choreoConnectPolicyDefinitionJsonContent != null) {
+                ccPolicyDefinition = new OperationPolicyDefinition();
+                ccPolicyDefinition.setContent(choreoConnectPolicyDefinitionJsonContent);
+                ccPolicyDefinition.setGatewayType(OperationPolicyDefinition.GatewayType.ChoreoConnect);
+                ccPolicyDefinition.setMd5Hash(APIUtil.getMd5OfOperationPolicyDefinition(ccPolicyDefinition));
+                operationPolicyData.setCcPolicyDefinition(ccPolicyDefinition);
+            }
+
+            operationPolicyData.setMd5Hash(APIUtil.getMd5OfOperationPolicy(operationPolicyData));
+
+            OperationPolicyData existingPolicy = apiProvider.getCommonOperationPolicyByPolicyName(
+                    policySpecification.getName(), policySpecification.getVersion(), organization,
+                    false);
+            String policyID = null;
+            if (existingPolicy == null) {
+                policyID = apiProvider.addCommonOperationPolicy(operationPolicyData, organization);
+                if (log.isDebugEnabled()) {
+                    log.debug("A common operation policy has been added with name " + policySpecification.getName());
+                }
+            } else {
+                throw new APIManagementException("Existing common operation policy found for the same name.");
+            }
+            operationPolicyData.setPolicyId(policyID);
+            OperationPolicyDataDTO createdPolicy = OperationPolicyMappingUtil.fromOperationPolicyDataToDTO(
+                    operationPolicyData);
+
+            return createdPolicy;
+        } catch (APIManagementException e) {
+            String errorMessage = "Error while adding a common operation policy." + e.getMessage();
+            throw new APIManagementException(errorMessage, ExceptionCodes.from(
+                    ExceptionCodes.OPERATION_POLICY_ALREADY_EXISTS,
+                    policySpecification.getName(), policySpecification.getVersion()));
+        } catch (Exception e) {
```

Comment:
Why are we capturing the general Exception class here? I think this will capture the APIManagementException throw in the 899 line and rewrap it 904. Normally, it is not a good practice to capture the general Exception. Shall we capture the specific Exceptions?

Reply:
Yes, this should be an IOException. I wonder how this returns a `409` if the policy exists since the thrown exception will get captured by `902` Exception and then it throws an `Internal Server Error`

Reply:
Should be an IOException

Reply:
Fixed by [b7f664a](https://github.com/wso2/carbon-apimgt/pull/11325/commits/b7f664af4e393bb1fe5d307dd9dc9c9107e372d6)